tests/test_uniswapv2.py

```diff
@@ -0,0 +1,148 @@
+from pymaker.model import Token
+import unittest
+
+@unittest.skip("TestUniswapV2 testing skipping")
+class TestUniswapV2(Contract):
+
+    pair_abi = Contract._load_abi(__name__, '../pyexchange/abi/IUniswapV2Pair.abi')
+    router_abi = Contract._load_abi(__name__, '../pyexchange/abi/IUniswapV2Router02.abi')
+    router_bin = Contract._load_bin(__name__, '../pyexchange/abi/IUniswapV2Router02.bin')
+    factory_abi = Contract._load_abi(__name__, '../pyexchange/abi/IUniswapV2Factory.abi')
+    factory_bin = Contract._load_bin(__name__, '../pyexchange/abi/IUniswapV2Factory.bin')
+
+    def setup_method(self):
+        py_evm_main.GENESIS_GAS_LIMIT = 10000000
+        self.web3 = Web3(EthereumTesterProvider(EthereumTester(PyEVMBackend())))
+        self.web3.eth.defaultAccount = self.web3.eth.accounts[0]
+        self.our_address = Address(self.web3.eth.defaultAccount)
+
+        self.router_address = self._deploy(self.web3, self.router_abi['abi'], self.router_bin, [])
+        self.factory_address = self._deploy(self.web3, self.factory_abi['abi'], self.factory_bin, [])
+        self._router_contract = self._get_contract(self.web3, self.router_abi['abi'], self.router_address)
+        self._factory_contract = self._get_contract(self.web3, self.factory_abi['abi'], self.factory_address)
+
+        print(self.router_address, self.factory_address, __name__)
+
+        # self.dai_token = ERC20Token(web3=self.web3, address=deploy_contract(self.web3,abi 'DAIToken'))
+        # self.token_transfer_proxy_address = deploy_contract(self.web3, 'TokenTransferProxy')
+        # self.exchange = ZrxExchange.deploy(self.web3, self.zrx_token.address, self.token_transfer_proxy_address)
+        # TODO: deploy the other uniswap v2 contracts
+        # self.web3.eth.contract(abi=json.loads(pkg_resources.resource_string('pymaker.deployment', f'abi/TokenTransferProxy.abi')))(address=self.token_transfer_proxy_address.address).functions.addAuthorizedAddress(self.exchange.address.address).transact()
+
+        # self.dgx = DSToken.deploy(self.web3, 'DGX')
+        self.dai = DSToken.deploy(self.web3, 'DAI')
+
+        # Kovan token addresses
+        self.token_a = Token("DAI", Address("0x4f96fe3b7a6cf9725f59d353f723c1bdb64ca6aa"), 18)
+        self.token_b = Token("USDC", Address("0x198419c5c340e8de47ce4c0e4711a03664d42cb2"), 6)
+        self.uniswap = UniswapV2(self.web3, self.token_a, self.token_b)
+
+    def test_get_block(self):
+        assert isinstance(self.uniswap.get_block(), int)
+
+    def test_approval(self):
+        # given
+        assert self.dai.allowance_of(self.our_address, self.router_address) == Wad(0)
+
+        # when
+        # self._router_contract.approve([self.dai], directly())
+        approval_function = directly()
+        
+        approval_function(self.dai, self.router_address, 'IUniswapV2Router02')
+
+        # then
+        assert self.dai.allowance_of(self.our_address, self.router_address) > Wad(0)
+
+    def test_getting_balances(self):
```

Comment:
As written, this test merely ensures no exception is raised.

Reply:
Updated test